### PR TITLE
Simplify webagg blitting.

### PR DESCRIPTION
**Pull request by anntzer**
- Directly reuse FigureCanvasAgg.get_renderer.  We don't need to init
  _last_buff there as we can just use buff.copy() in get_diff_image()
  (If we really care about reusing the same array whenever possible, we
  could add a shape check before the removed copyto() and reuse the old
  array if the shapes match, but I doubt it matters.)  Instead, init it
  once to a non-valid shape in the constructor.

  Note that the comments in the function were outdated since 78c182d.

- First creating `pixels` and then viewing it as `buff` is slighly
  simpler than doing it the other way round.

The animation example in #19059 works for me.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Review by tacaswell — APPROVED** [review]

**Review by tacaswell — APPROVED** [review]

File: lib/matplotlib/backends/backend_webagg_core.py [hunk 1]
```diff
@@ -227,17 +230,17 @@ def get_diff_image(self):
...
-                    .reshape((renderer.height, renderer.width)))
-
-            # If any pixels have transparency, we need to force a full
-            # draw as we cannot overlay new on top of old.
-            pixels = buff.view(dtype=np.uint8).reshape(buff.shape + (4,))
+            buff = pixels.view(np.uint32).squeeze(2)
 
-            if self._force_full or np.any(pixels[:, :, 3] != 255):
+            if (self._force_full
+                    or buff.shape != self._last_buff.shape
```
Comment (ianhi) [inline lib/matplotlib/backends/backend_webagg_core.py:240]: may be nice to add a comment as well about this line. It took me a while to hunt down why it was ok to compare the initial `np.empty((0,0))` with the buffer and this line is the escape

**Review by ianhi — APPROVED** [review]
nice! much simpler

File: lib/matplotlib/backends/backend_webagg_core.py (hunk 1 above)
Comment (tacaswell) [inline lib/matplotlib/backends/backend_webagg_core.py:240]:
```suggestion
                    # If the buffer has changed size we need to do a full draw
                    or buff.shape != self._last_buff.shape
```

**Comment by anntzer** [comment]
Added comment as requested.

**Pull request merged**